packages/yoroi-extension/app/ergo-connector/components/signin/CardanoSignTxPage.js

```diff
@@ -250,17 +299,135 @@ class SignTxPage extends Component<Props> {
+          )
+          return <p>{address}</p>
+        })}
+      </div>
+    )
+  }
+
+
   render(): Node {
     const { form } = this;
     const walletPasswordField = form.$('walletPassword');
 
     const { intl } = this.context;
     const { txData, onCancel, } = this.props;
+    const { showUtxoDetails } = this.state
     return (
       <>
         <ProgressBar step={2} />
         <div className={styles.component}>
-          <div>
+          {
+            !showUtxoDetails ?(
+              <div>
+                <div>
+                  <h1 className={styles.title}>{intl.formatMessage(messages.title)}</h1>
+                </div>
+                <div className={styles.transactionWrapper}>
+                  <p className={styles.transactionId}>
+                    {intl.formatMessage(messages.receiver)}
+                  </p>
+                  <p className={styles.hash}>{this.renderAddresses()}</p>
+                  <button onClick={() => this.toggleUtxoDetails(true)} type='button' className={styles.utxo}>
+                    <p>{intl.formatMessage(messages.txDetails)}</p>
+                    <ArrowRight />
+                  </button>
+                </div>
+                <div className={styles.info}>
+                  <div className={styles.infoRaw}>
+                    <p className={styles.label}>{intl.formatMessage(globalMessages.amount)}</p>
+                    <p className={styles.labelValue}>
+                      {this.renderAmountDisplay({
+                        entry: {
+                          identifier: txData.fee.tokenId,
+                          networkId: txData.fee.networkId,
+                          amount: (new BigNumber(txData.fee.amount)).negated(),
+                        },
+                    })}
+                    </p>
+                  </div>
+                  <div className={styles.infoRaw}>
+                    <p className={styles.label}>{intl.formatMessage(globalMessages.feeLabel)}</p>
+                    <p className={styles.labelValue}>
+                      {this.renderAmountDisplay({
+                  entry: {
+                    identifier: txData.fee.tokenId,
+                    networkId: txData.fee.networkId,
+                    amount: (new BigNumber(txData.fee.amount)).negated(),
+                  },
+                })}
+                    </p>
+                  </div>
+                  <div className={styles.totalAmoundCard}>
+                    <p className={styles.totalAmoundLable}>
+                      {intl.formatMessage(globalMessages.walletSendConfirmationTotalLabel)}
+                    </p>
+                    <p className={styles.totalAmound}>
+                      {this.renderAmountDisplay({
+                  entry: {
+                    identifier: txData.fee.tokenId,
+                    networkId: txData.fee.networkId,
+                    amount: (new BigNumber(txData.fee.amount)).negated(),
+                  },
+                })}
```

Comment:
@yushih  Note we are displaying the fees inside the `total amount ` card as a placeholder. the tx data is not in the right format to be calculated!

Reply:
@yushih , can you plz take a look here. When transaction is parsed with the serialization-lib there should be a `.fee()` function available that returns the `BigNum`, so it should be possible to do something like:
```
const feeLovelacesStr = parsedTx.fee().to_str();
```